cpp/src/io/orc/writer_impl.cu

```diff
@@ -72,6 +72,17 @@ namespace orc {
 using namespace cudf::io::orc;
 using namespace cudf::io;
 
+template <typename T>
+[[nodiscard]] constexpr int varint_size(T val)
```

Comment:
Could not eliminate the duplicated implementation without creating a new header file. Still, moved one of the implementations from a header to a source file.
This function should be in a utility header and reused in ORC and Parquet.